pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -798,14 +800,35 @@ func PruneEmpty(config map[string]interface{}) {
 	}
 }
 
-func addAddresses(config map[string]interface{}, machine *capi.Machine) {
+func addAddresses(secrets corecontrollers.SecretCache, config map[string]interface{}, machine *capi.Machine) error {
+	if data := machine.Annotations[InternalAddressAnnotation]; data != "" {
+		config["node-ip"] = data
+	}
+
 	if data := machine.Annotations[AddressAnnotation]; data != "" {
 		config["node-external-ip"] = data
+		// Don't check for an IP address from the secret if it is already provided.
+		return nil
 	}
 
-	if data := machine.Annotations[InternalAddressAnnotation]; data != "" {
-		config["node-ip"] = data
+	if machine.Annotations["objectset.rio.cattle.io/id"] == "unmanaged-machine" {
+		// Unmanaged machines won't have a state secret
+		return nil
+	}
+
+	if secret, err := secrets.Get(machine.Spec.InfrastructureRef.Namespace, name.SafeConcatName(machine.Spec.InfrastructureRef.Name, "machine", "state")); err != nil {
+		return err
+	} else if secret != nil && len(secret.Data["extractedConfig"]) != 0 {
+		if driverConfig, err := nodeconfig.ExtractConfigJSON(base64.StdEncoding.EncodeToString(secret.Data["extractedConfig"])); err != nil {
+			return err
+		} else if len(driverConfig) != 0 && len(convert.ToString(values.GetValueN(driverConfig, "Driver", "IPAddress"))) != 0 {
+			config["node-external-ip"] = convert.ToString(values.GetValueN(driverConfig, "Driver", "IPAddress"))
+		}
```

Comment:
Is it OK we do nothing if the condition from line 824 is not met (so no error and no `"node-external-ip"` set)? The error return on line 828 does not cover this.